### PR TITLE
VZ-8137: Add analysis support for innoDBCluster and pods hanging on termination and for pods waiting on readiness gates

### PR DESCRIPTION
**Pull request by George-Aeillo**
In this PR, the analysis tool is expanded to detect whether a pod has been terminating for 10 minutes or greater and is waiting on readiness gates. This PR also allows the bug-report command to capture innoDBCluster resources along with the analysis of innoDBCluster resources to see if it has been terminating for 10 minutes or greater. This PR also changes some of the termination detection and reporting for namespaces to keep consistent with innoDBCluster and pod resources.

### CODE REVIEW

**Review by mgianatagh — APPROVED** [review]

**Review by agv426 — APPROVED** [review]

File: tools/vz/pkg/analysis/internal/util/cluster/mysql.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	return nil
+}
+
+// getInnoDBClusterResource returns the InnoDBCluster list that is in the inno-db-cluster.json file
+func getInnoDBClusterResources(log *zap.SugaredLogger, path string) (innoDBClusterObject *unstructured.UnstructuredList, err error) {
+	resourceToReturn := unstructured.UnstructuredList{}
+	file, err := os.Open(path)
+	defer file.Close()
+	if err != nil {
+		log.Debug("file %s not found", path)
```
Comment (marcogelle) [inline tools/vz/pkg/analysis/internal/util/cluster/mysql.go:64]: should be `log.Debugf`

File: tools/vz/pkg/analysis/internal/util/cluster/mysql.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		return nil, nil
+	}
+	fileBytes, err := io.ReadAll(file)
+	if err != nil {
+		log.Error("Failed reading namespace.json file %s", path)
```
Comment (marcogelle) [inline tools/vz/pkg/analysis/internal/util/cluster/mysql.go:69]: should be `log.Errorf`

File: tools/vz/pkg/analysis/internal/util/cluster/mysql.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		return nil, err
+	}
+	err = resourceToReturn.UnmarshalJSON(fileBytes)
+	if err != nil {
+		log.Error("Failed to unmarshal namespace resource at %s", path)
```
Comment (marcogelle) [inline tools/vz/pkg/analysis/internal/util/cluster/mysql.go:74]: should be `log.Errorf`

File: tools/vz/pkg/analysis/internal/util/cluster/mysql.go (hunk 1 above)
Comment (George-Aeillo) [reply tools/vz/pkg/analysis/internal/util/cluster/mysql.go]: Ok! Thanks!

File: tools/vz/pkg/analysis/internal/util/cluster/mysql.go (hunk 2 above)
Comment (George-Aeillo) [reply tools/vz/pkg/analysis/internal/util/cluster/mysql.go]: Ok! Thanks!

File: tools/vz/pkg/analysis/internal/util/cluster/mysql.go (hunk 3 above)
Comment (George-Aeillo) [reply tools/vz/pkg/analysis/internal/util/cluster/mysql.go]: Ok! Thanks!

**Review by Abhijithang — APPROVED** [review]

**Pull request merged**